libvast/builtins/connectors/file.cpp

```diff
@@ -0,0 +1,262 @@
+                                 fmt::format("could not parse duration: {}",
+                                             args[i + 1]));
+        }
+      } else if (arg == "--uds") {
+        file_type = std::filesystem::file_type::socket;
+      } else if (arg == "-" || arg == "stdin") {
+        path = stdin_path;
+      } else if (arg == "-f" || arg == "--follow") {
+        following = true;
+      } else if (not arg.starts_with("-")) {
+        std::error_code err{};
+        auto status = std::filesystem::status(arg, err);
+        if (err) {
+          return caf::make_error(ec::parse_error,
+                                 fmt::format("could not parse file path {}: {}",
+                                             arg, err));
+        }
+        auto exists = (status.type() != std::filesystem::file_type::not_found);
+        if (not exists) {
+          return caf::make_error(ec::filesystem_error,
+                                 fmt::format("file {} does not exist", arg));
+        }
+        if (not path.empty()) {
+          return caf::make_error(ec::syntax_error,
+                                 fmt::format(
+                                   "multiple paths have been "
+                                   "specified; currently only one "
+                                   "path per read operator call is valid"));
+        }
+        auto permissions = status.permissions();
+        auto readable = ((permissions & std::filesystem::perms::owner_read)
+                           != std::filesystem::perms::none
+                         && (permissions & std::filesystem::perms::group_read)
+                              != std::filesystem::perms::none
+                         && (permissions & std::filesystem::perms::others_read)
+                              != std::filesystem::perms::none);
+        if (not readable) {
+          return caf::make_error(ec::filesystem_error,
+                                 fmt::format("file {} is not readable", arg));
+        }
+        path = arg;
+      }
+    }
+    if (path.empty()) {
+      return caf::make_error(ec::syntax_error,
+                             fmt::format("no file specified"));
+    }
+    auto fd = STDIN_FILENO;
+    switch (file_type) {
+      default:
+        return caf::make_error(ec::filesystem_error, "unsupported file type",
+                               path);
+      case std::filesystem::file_type::socket: {
+        if (path == stdin_path) {
+          return caf::make_error(ec::filesystem_error,
+                                 "cannot use STDIN as UNIX "
+                                 "domain socket");
+        }
+        auto uds = detail::unix_domain_socket::connect(path);
+        if (!uds) {
+          return caf::make_error(ec::filesystem_error,
+                                 "failed to connect to UNIX domain socket at",
+                                 path);
+        }
+        fd = uds.recv_fd(); // Blocks!
```

Comment:
I'm not familiar with the exact semantics around this. Is the call to `recv_fd` problematic here?

Reply:
`recv_fd()` blocks until it receives the file descriptor from the socket.
I can maybe add a polling with timeout.

Reply:
I tried to understand what happens here, and either I don't get it, or this is not all what we want to do. What about `fd = uds.fd`?

Reply:
I studied the `uds` code a bit, and I think `uds.fd` should suffice indeed. https://github.com/tenzir/vast/pull/3088/commits/dea1b9f7a58896bc73b8ca5a0a124eeae5629cc2